lib/command/run.rb

```diff
@@ -51,59 +69,89 @@ class Run < Base
       ```
     EX
 
-    attr_reader :location, :workload_to_clone, :workload_clone, :container
+    INTERACTIVE_COMMANDS = [
+      "bash",
+      "rails console",
+      "rails c",
+      "rails dbconsole",
+      "rails db"
+    ].freeze
+
+    attr_reader :interactive, :detached, :location, :workload_to_clone, :workload_clone, :container, :command
 
     def call # rubocop:disable Metrics/MethodLength
+      @interactive = config.options[:interactive] || interactive_command?
+      @detached = config.options[:detached]
+
       @location = config.location
-      @workload_to_clone = config.options["workload"] || config[:one_off_workload]
+      @workload_to_clone = config.options[:workload] || config[:one_off_workload]
       @workload_clone = "#{workload_to_clone}-run-#{random_four_digits}"
 
-      step("Cloning workload '#{workload_to_clone}' on app '#{config.options[:app]}' to '#{workload_clone}'") do
+      step("Cloning workload '#{workload_to_clone}' on app '#{config.app}' to '#{workload_clone}'") do
         clone_workload
       end
 
       wait_for_workload(workload_clone)
+      start_cron_workload(workload_clone, location)
       wait_for_replica(workload_clone, location)
-      run_in_replica
+      if interactive
+        run_interactive
+      else
+        run_non_interactive
+      end
     ensure
-      progress.puts
-      ensure_workload_deleted(workload_clone)
+      if interactive
+        progress.puts
+        ensure_workload_deleted(workload_clone)
+      end
     end
 
     private
 
-    def clone_workload # rubocop:disable Metrics/MethodLength
-      # Create a base copy of workload props
+    def interactive_command?
+      INTERACTIVE_COMMANDS.include?(args_join(config.args))
+    end
+
+    def clone_workload # rubocop:disable Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+      # Get base specs of workload
       spec = cp.fetch_workload!(workload_to_clone).fetch("spec")
       container_spec = spec["containers"].detect { _1["name"] == workload_to_clone } || spec["containers"].first
       @container = container_spec["name"]
 
-      # remove other containers if any
+      # Remove other containers if any
       spec["containers"] = [container_spec]
 
-      # Stub workload command with dummy server that just responds to port
-      # Needed to avoid execution of ENTRYPOINT and CMD of Dockerfile
-      container_spec["command"] = "ruby"
-      container_spec["args"] = ["-e", Scripts.http_dummy_server_ruby]
+      # Set runner
+      container_spec["env"] ||= []
+      container_spec["env"] << { "name" => "CONTROLPLANE_RUNNER", "value" => runner_script }
+      if interactive
+        container_spec["command"] = "tail"
+        container_spec["args"] = ["-f", "/dev/null"]
+        @command = %(bash -c 'eval "$CONTROLPLANE_RUNNER"')
```

Comment:
Using `tail -f /dev/null` as the default command for interactive mode allows the cron workload to stay active indefinitely, so that we can connect to it later and run the script using `cpln workload exec`.